### PR TITLE
Bugfix/flow 11052 radios do not play

### PR DESCRIPTION
**Pull request by lmancel**
Extract mp4 parsing from ipcStreamer. Choose parser depending on the type of ffmpeg's output:
- for radios (mp3): send data at a specified interval
- for tv (mp4): send data whenever a "moof"/"mdat" segment is complete

### CODE REVIEW

File: src/frontend/src/players/parsers/mp4.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  private initSegmentComplete = false
+
+  private appendToInitSegment(chunk: Buffer): void {
+    this.initSegment = this.initSegment ? Buffer.concat([this.initSegment, chunk]) : chunk
+  }
+
+  /**
+   * From a chunk of data, attempt to find a "moof" box
+   * This box type announces the beginning of a new data segment
+   * This allows to structure forwarded data to somehthing that can be handled easily
```
Comment (phil1618) [inline src/frontend/src/players/parsers/mp4.ts]:
```suggestion
   * This allows to structure forwarded data to something that can be handled easily
```

File: src/frontend/src/players/parsers/mp4.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    // Attempt to find a "moof" box in the chunk
+    // if it is found, it means we reached the end of the previous box's data
+    const [moof] = mp4.tools.findBox(chunk, ['moof'])
+
+    if (!this.initSegmentComplete && moof) {
+      // receiving a "moof" means that the initialization segment is complete
+      this.initSegmentComplete = true
+
+      /**
+       * The call to "findBox" above returns the requested boxes' content if found, without their header
```
Comment (phil1618) [inline src/frontend/src/players/parsers/mp4.ts:34]:
```suggestion
       * The call to "findBox" above returns the requested boxe's content if found, without their header
```

File: src/keyboard/keyboardController.ts [hunk 3]
```diff
@@ -128,7 +128,7 @@ class Keyboard {
       parent.on('close', callback)
       this.parentCloseCallback = { parent, callback }
     }
-    this.keyboardWindow.setParentWindow(parent)
+    this.keyboardWindow?.setParentWindow(parent)
```
Comment (phil1618) [inline src/keyboard/keyboardController.ts:131]: Thanks !

**Review by phil1618 — APPROVED** [review]

File: src/frontend/src/players/parsers/mp4.ts (hunk 2 above)
Comment (lmancel) [reply src/frontend/src/players/parsers/mp4.ts:34]: "boxes" is plural here ("findBox" receives an array of box name), it is written as intended

File: src/frontend/src/players/pipelines/ffmpegPlayingPipeline.ts [hunk 4]
```diff
@@ -171,7 +177,7 @@ class PlayingPipeline extends StateMachineImpl<PipelineState> {
 
   kill(): void {
     this.command?.unpipe(this.streamer)
-    this.command?.kill('SIGKILL')
```
Comment (Azevedo-Dani) [inline src/frontend/src/players/pipelines/ffmpegPlayingPipeline.ts:174]: Are you sure it's a good idea?
I'm afraid, we'll have the same issue as before, the process isn't killed in Linux or Windows, and sometimes we have to specify the Kill level.

Comment (lmancel) [reply src/frontend/src/players/pipelines/ffmpegPlayingPipeline.ts:174]: "this.command" is not an FfmpegCommand anymore. It has simply been moved here -> https://github.com/taktik/flowr-desktop/pull/63/files/b676c187bac3b9fbe4aa3dad43a540082b3543e6#diff-e849dcaa6e5d73a418062757f27427a77e3786fd950a6d0ef6f72199e9daa1b9R81

Comment (Azevedo-Dani) [reply src/frontend/src/players/pipelines/ffmpegPlayingPipeline.ts:174]: my bad :)

**Review by Azevedo-Dani — APPROVED** [review]

**Pull request merged**